google-cloud-bigtable/src/main/java/com/google/cloud/bigtable/data/v2/stub/DynamicFlowControlCallable.java

```diff
@@ -0,0 +1,170 @@
+/*
+ * Copyright 2021 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.google.cloud.bigtable.data.v2.stub;
+
+import com.google.api.core.ApiFuture;
+import com.google.api.gax.batching.FlowControlEventStats;
+import com.google.api.gax.batching.FlowController;
+import com.google.api.gax.rpc.ApiCallContext;
+import com.google.api.gax.rpc.UnaryCallable;
+import com.google.common.base.Stopwatch;
+import com.google.common.util.concurrent.MoreExecutors;
+import java.util.concurrent.TimeUnit;
+import javax.annotation.Nonnull;
+
+/**
+ * A callable that records rpc latency and adjusts flow control thresholds for latency based
+ * throttling.
+ */
+final class DynamicFlowControlCallable extends UnaryCallable {
+  private final FlowController flowController;
+  private final FlowControlEventStats flowControlEvents;
+  private final DynamicFlowControlStats dynamicFlowControlStats;
```

Comment:
I think we can improve on class naming here. The names are too similar which makes hard to remember what each does. Unfortunately I dont have good suggestions :(

Reply:
Yeah, I noticed this too but also don't have any better idea 😢 
But after flowControlEvents is moved to flowController I think it's less confusing now?